src/earn/EarnCard.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { View } from 'react-native'
+import ItemSeparator from 'src/components/ItemSeparator'
+import EarnActivePool from 'src/earn/EarnActivePool'
+import EarnCta from 'src/earn/EarnCta'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import { Spacing } from 'src/styles/styles'
+import { useTokenInfo } from 'src/tokens/hooks'
+
+interface Props {
+  depositTokenId: string
+  poolTokenId: string
+}
+
+export function EarnCardDiscover({ depositTokenId, poolTokenId }: Props) {
+  const showStablecoinEarn = getFeatureGate(StatsigFeatureGates.SHOW_STABLECOIN_EARN)
+  const poolToken = useTokenInfo(poolTokenId)
+
+  if (showStablecoinEarn) {
+    return poolToken && poolToken.balance.gt(0) ? (
+      <EarnActivePool
+        buttonDisplay="ExitAndDeposit"
+        depositTokenId={depositTokenId}
+        poolTokenId={poolTokenId}
+      />
+    ) : (
+      <EarnCta />
+    )
+  }
+  return null
+}
+
+export function EarnCardTokenDetails({ depositTokenId, poolTokenId }: Props) {
+  const showStablecoinEarn = getFeatureGate(StatsigFeatureGates.SHOW_STABLECOIN_EARN)
+  const poolToken = useTokenInfo(poolTokenId)
+
+  if (showStablecoinEarn) {
+    return poolToken && poolToken.balance.gt(0) ? (
```

Comment:
nit: could probably combine these in a single if, since the fallback for both is `null`

Reply:
Changed in 4b31a0a11181ee7543001a2d6f71485ecc461f9e!